api/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -56,15 +56,37 @@ type OpenstackDataPlaneServiceCert struct {
 	EDPMRoleServiceName string `json:"edpmRoleServiceName,omitempty"`
 }
 
+// OpenStackDataPlaneConfigMap defines the properties of the ConfigMaps used
+// by the service.
+type OpenStackDataPlaneConfigMap struct {
```

Comment:
One thing we can think about is if we want to create something that is more similar how the volume in k8s expresses the CM/Secret sources. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#configmapvolumesource-v1-core

Copying that patter would mean something like:
```golang

// this type needs a validation that only one of the fields can be non nil
type ServiceDataSource struct {
    ConfigMap *ConfigMapRef
    Secret *SecretRef
}

type ConfigMapRef struct {
    Name string
    Optional bool
}

type SecretRef struct {
    Name string
    Optional bool
}
//...

type OpenStackDataPlaneServiceSpec struct {
    // then we can have a single list instead of two lists
    dataSources []ServiceDataSource
}
```

* This would make the interface more familiar for a k8s user. 
* Adding different data source types later would be a bit more structured
* maybe it represents other hard earned knowledge of the k8s API designers but I'm not sure.

Overall this is not a big difference so not doing this is also OK to me now.

Reply:
I've combined this suggestion with the one from @fao89 to use the same struct as AnsibleVarsFrom, and refactored the common code. I've also left the existing ConfigMaps/Secrets fields, so that we can merge this with backwards compatibility, then clean those up after migrating cifmw/install_yamls/etc.

Reply:
the new struct looks OK to me. Also agree about the backward compatible change strategy.